pkg/exporter/sumologicexporter/exporter.go

```diff
@@ -137,6 +138,40 @@ func initExporter(cfg *Config, createSettings component.ExporterCreateSettings)
 	return se, nil
 }
 
+// addSourceMetadataFields adds source related attribute names to the list of
+// metadata attributes.
+func addSourceMetadataFields(metadataAttributes []string) []string {
+	var (
+		sourceCategory bool
+		sourceHost     bool
+		sourceName     bool
+	)
+
+	for _, attr := range metadataAttributes {
+		switch attr {
+		case attributeKeySourceCategory:
+			sourceCategory = true
+		case attributeKeySourceHost:
+			sourceHost = true
+		case attributeKeySourceName:
+			sourceName = true
+		default:
+		}
+	}
+
+	if !sourceCategory {
+		metadataAttributes = append(metadataAttributes, attributeKeySourceCategory)
+	}
+	if !sourceHost {
+		metadataAttributes = append(metadataAttributes, attributeKeySourceHost)
+	}
+	if !sourceName {
+		metadataAttributes = append(metadataAttributes, attributeKeySourceName)
+	}
+
+	return metadataAttributes
+}
```

Comment:
cannot we just add element to the list?
is it any risk related to duplication?

Reply:
I don't believe there's risk of duplicating anything in terms of headers/attributes but we do it once here and n times ( where n is the number of metadata attribute regexes) for each processed entry right? So it might be worth the hassle here